File: it.unibo.boundaryWalk/app/src/main/java/it/unibo/annotatedExample/UniboAnnotationUtil.java
package it.unibo.annotatedExample;
import java.lang.annotation.Annotation;
import java.lang.reflect.AnnotatedElement;
import java.lang.reflect.Method;

public class UniboAnnotationUtil {
    public static String getActorName( AnnotatedElement element ){
        try {
            Annotation[] classAnnotations = element.getAnnotations();
            for (Annotation annotation : classAnnotations) {
                if (annotation instanceof ISSActorSpec) {
                    ISSActorSpec info = (ISSActorSpec) annotation;
                    return  info.actorName();
                }
            }
        } catch (Exception exception) {
            exception.printStackTrace();
        }
        return "error about actorName";
    }


    public static void initializeObject(Object object)   {
        //println("initializeObject object=" + object);
        Class<?> clazz = object.getClass();
        for (Method method : clazz.getDeclaredMethods()) {
            //System.out.println("initializeObject method=" + method);
            if (method.isAnnotationPresent(InitSpec.class)) {
                method.setAccessible(true);
                try{
                    //System.out.println("initializeObject invoke " + method);
                    method.invoke(object);
                }catch( Exception e ){
                    e.printStackTrace();
                }
            }
        }
    }

}

File: it.unibo.boundaryWalk/app/src/main/kotlin/it/unibo/annotExample/UniboAnnotationUtil.kt
package it.unibo.annotExample
import java.lang.reflect.AnnotatedElement

object UniboAnnotationUtil {
    fun getActorName(element: AnnotatedElement): String {
        try {
            val classAnnotations = element.annotations
            for (annotation in classAnnotations) {
                if (annotation is ISSActorSpec) {
                    return annotation.actorName
                }
            }
        } catch (exception: Exception) {
            exception.printStackTrace()
        }
        return "error about actorName"
    }

    fun initializeObject(obj : Any) {
        println("initializeObject object=" + obj)
        val clazz: Class<*> = obj.javaClass
        for (method in clazz.declaredMethods) {
            //println("initializeObject method=" + method);
            if (method.isAnnotationPresent(InitSpec::class.java)) {
                method.isAccessible = true
                try {
                    println("initializeObject invoke " + method);
                    method.invoke( obj )
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        }
    }
}